npm/pkg/dataplane/policies/chain-management_linux.go

```diff
@@ -345,29 +357,29 @@ func (pMgr *PolicyManager) creatorForBootup(currentChains map[string]struct{}) *
 	return creator
 }
 
-// add/reposition the jump from FORWARD chain to AZURE-NPM chain to be in the correct position based on config:
+// add/reposition the jump from a given chain to AZURE-NPM chain to be in the correct position based on config:
 // option 1) jump to AZURE-NPM chain should be the first rule
-// option 2) jump to AZURE-NPM chain should be after the jump to KUBE-SERVICES chain
-func (pMgr *PolicyManager) positionAzureChainJumpRule() error {
+// option 2) jump to AZURE-NPM chain should be after the jump to KUBE-SERVICES chain (this is the default for the OUTPUT chain for IPVS compatibility)
+func (pMgr *PolicyManager) positionAzureChainJumpRule(fromChain string) error {
 	// get the line number for the azure jump
-	azureChainLineNum, err := pMgr.chainLineNumber(util.IptablesAzureChain)
+	azureChainLineNum, err := pMgr.chainLineNumber(util.IptablesAzureChain, fromChain)
 	if err != nil {
-		baseErrString := "failed to get index of jump from FORWARD chain to AZURE-NPM chain"
+		baseErrString := "failed to get index of jump from " + fromChain + " chain to AZURE-NPM chain"
 		metrics.SendErrorLogAndMetric(util.IptmID, "error: %s: %s", baseErrString, err.Error())
 		return npmerrors.SimpleErrorWrapper(baseErrString, err)
 	}
 
-	if pMgr.PlaceAzureChainFirst == util.PlaceAzureChainFirst && azureChainLineNum == 1 {
+	if pMgr.PlaceAzureChainFirst == util.PlaceAzureChainFirst && azureChainLineNum == 1 && fromChain == util.IptablesForwardChain {
```

Comment:
this would actually change current behavior for placefirst from forward chain (as is, it could never take this branch)

Reply:
I'd suggest having a PlaceFirst argument to this function, and you can pass in pMgr.PlaceAzureChainFirst for FORWARD chain, and util.PlaceAfterKubeServices for OUTPUT